impf/alert.py

```diff
@@ -71,19 +98,19 @@ def zulip_read() -> str:
 
 
 def telegram_send(message: str) -> None:
-    api_token = settings.TELEGRAM_BOT_TOKEN
-    chat_id = settings.TELEGRAM_BOT_CHATID
-
-    url = f'https://api.telegram.org/bot{api_token}/sendMessage'
-    params = {
-        'chat_id': chat_id,
-        'parse_mode': 'Markdown',
-        'text': message
-    }
-
-    response = requests.get(url, params=params, headers=HEADERS)
-    logger.debug(response)
-
+    data = json.load(open('telegram.json', 'r'))
+    bot.send_message(data["message_id"],message)
+        
+def telegram_read() -> str:
+    data = json.load(open('telegram.json', 'r'))
+    msgs = bot.get_updates(data["offset"])
```

Comment:
We should work with `data.get('offset')` here to avoid an `IndexError` if the field is not present.

edit: Is it also possible to specify a time-based offset? This would eliminate the need for an extra file to store what I assume is a message_id-based offset.

Reply:
I will change it to your suggestion to avoid exceptions on missing fields.

The offset is the update offset. Description can be found here: [https://core.telegram.org/bots/api#getupdates](https://core.telegram.org/bots/api#getupdates)

Yes it is maybe a good idea to work with a time-based ignore threshold to ignore already processed messages.

Reply:
I think we can ignore the offset if we just get the last 10 messages (remember messages are read every 15s) and take the latest one matching the SMS pattern. We don't have to be so accurate here. But I cannot seem to find a straight forward method to *poll* messages periodically in the [message API docs](https://core.telegram.org/methods#working-with-messages).

side note: another reason we should avoid relying on a file is race conditions, again when using concurrency.